scripts/ub-check.py

```diff
@@ -0,0 +1,226 @@
+        "arm64 MacOS": [    f'{mainfile.split(".")[0]}.Clang.O0',
+                            f'{mainfile.split(".")[0]}.Clang.O2',
+                            f'{mainfile.split(".")[0]}.Clang.O3',
+                            f'{mainfile.split(".")[0]}.CPP14.GCC13.O0',
+                            f'{mainfile.split(".")[0]}.CPP14.GCC13.O2',
+                            f'{mainfile.split(".")[0]}.CPP14.GCC13.O3',
+                            f'{mainfile.split(".")[0]}.GNU20.GCC13.O0',
+                            f'{mainfile.split(".")[0]}.GNU20.GCC13.O2',
+                            f'{mainfile.split(".")[0]}.GNU20.GCC13.O3',
+                            f'{mainfile.split(".")[0]}.GCC13.O0',
+                            f'{mainfile.split(".")[0]}.GCC13.O2',
+                            f'{mainfile.split(".")[0]}.GCC13.O3',  ],
+    }
+    compile_commands = compile_commands_dict[runs_on]
+    compile_products = compile_products_dict[runs_on]
+
+    return_status = {}
+    this_file_looks_odd = False
+    for compile_command, compile_product in zip(compile_commands, compile_products):
+        print(compile_command, end=' ')
+        result = subprocess.run(compile_command, shell=True)
+        if result.returncode != 0:
+            print(f'{RED}CE({result.returncode}){RESET}')
+            this_file_looks_odd = True
+            status_vector = [CE(result.returncode)]
+        else: 
+            status_vector = [CompileOK()]
+            print(f'{GREEN}OK{RESET}')
+
+            for e in examples:
+                print(f'{compile_product} < {e} > {e.replace(".in", ".out")}', end=' ')
+                with open(e, 'r') as fstdin:
+                    with open(e.replace(".in", ".out"), 'w') as fstdout:
+                        result = subprocess.run(f'{os.path.join(os.path.curdir, compile_product)}', stdin=fstdin, stdout=fstdout, shell=True)
+                if result.returncode != 0:
+                    print(f'{RED}RE({result.returncode})){RESET}')
+                    this_file_looks_odd = True
+                    status_vector.append(RE(result.returncode))
+                else:
+                    print(f'{GREEN}OK{RESET}')
+                    print(f'diff -b -B {e.replace(".in", ".out")} {e.replace(".in", ".ans")}', end=' ')
+                    result = subprocess.run(f'diff -b -B {e.replace(".in", ".out")} {e.replace(".in", ".ans")}', shell=True)
+                    if result.returncode != 0:
+                        print(f'{RED}WA{RESET}')
+                        this_file_looks_odd = True
+                        status_vector.append(WA(result.returncode))
+                    else:
+                        print(f'{GREEN}AC{RESET}')
+                        status_vector.append(AC())
+        print(f'{compile_product.split(os.path.pathsep)[-1]}: ', end='')
+        for _ in status_vector:
+            print(f'{_.color}{_.__class__.__name__}({_.errcode}){RESET}', end='; ')
```

Comment:
It may be better to implement `__str__()`[^1].

[^1]: https://docs.python.org/3/reference/datamodel.html#object.__str__

Reply:
Indeed. But it's a problem that the console and file needs two string-ify methods (with or w/o colors). This issue is killing me lately

Reply:
PTAL Thanks! https://github.com/OI-wiki/OI-wiki/actions/runs/10738883306